src/api/app/models/bs_request_action.rb

```diff
@@ -920,6 +920,8 @@ def expand_targets(ignore_build_state)
   end
 
   def source_access_check!
+    return if Project.is_remote_project?(source_project, true)
```

Comment:
You can't hide the remote instance for certain users anymore by disabling access of the project which defines the remote instance when doing this.

To complete my former statement: It is up to the remote instance to deliver a package source or not. But this may still happen based on the credentials which are used to access in the remote instance. And they would be part of the local project defining the remote instance.

Eg.

   <remoteurl>https://adrian:secret@api.opensuse.org/public</remoteurl>